### ISSUE TITLE
Access non-editable fields in form with `fields_for_collection`

### ISSUE THREAD

**Issue by nerdinand** [hanami/helpers#121]
In my application I'm building a big form that edits multiple records at once. I'm doing this by building an html table where each row corresponds to a record. Some of the attributes of my records are not supposed to be editable and should just be displayed (e.g. `id`). 

It seems that there is no official solution at the moment to access and display that attribute. In my code I have to resort to using `_value` which is part of the private API to achieve this.

Here's an example:
```haml
= form_for :thing, routes.thing_path(thing.id), method: :patch, values: {thing: thing} do
  - table class: 'table table-striped' do
    - tbody do
      - fields_for_collection :sub_entities do
        - tr do
          - td do
            - text(_value(:id))
          - td do
            - text_field(:attribute)
```

Is there a better solution or should I do it this way?

**Comment by nerdinand** [comment]
Related: Is there a way of accessing the current instance the form is dealing with? For example if I wanted to access the entire `sub_entity` for the current table row, not just its `id`?

**Comment by nerdinand** [comment]
Here's a way of accessing `sub_entity` I came up with, but it's insane:
```
@values.get(:thing, :sub_entities)[@name.match(/\d+/)[-1].to_i]
```

**Comment by AlfonsoUceda** [comment]
@nerdinand how is the expected form in html? To help us to understand better, thanks.

**Comment by nerdinand** [comment]
The desired HTML looks something like this:

```html
<form action="/thing/1" method="POST" accept-charset="utf-8" id="thing-form">
  <table class="table table-striped">
    <tbody>
      <tr> <!-- One tr for each instance of :sub_entities -->
        <td>
          1 <!-- id of sub_entity -->
        </td>
        <td>
          <input type="text" name="thing[sub_entities][][attribute]" id="thing-sub-entity-0-attribute" value="value of attribute">
        </td>
      </tr>
      <tr> <!-- One tr for each instance of :sub_entities -->
        <td>
          2 <!-- id of sub_entity -->
        </td>
        <td>
          <input type="text" name="thing[sub_entities][][attribute]" id="thing-sub-entity-1-attribute" value="value of attribute">
        </td>
      </tr>
      <!-- many more rows, one for each entry in :sub_entities -->
    </tbody>
  </table>
</form>
```

To give the user an indication of which element they're dealing with, the first column shows a value specific to the `sub_entity` instance. In this case it's the `id`. Here, `_value(:id)` suffices to get that value, though I doubt I'm supposed to use that method anywhere...

This is only the simplest case. In another case, I want to pass the `sub_entity` we're dealing with to a custom view helper which renders something more complex. That's why I need access to the `sub_entity` object. Hence this crazy hack to access the `sub_entity` object: `@values.get(:thing, :sub_entities)[@name.match(/\d+/)[-1].to_i]`.

As this deals with potentially altering the API design, maybe @jodosha should chime in?

**Comment by AlfonsoUceda** [comment]
@nerdinand Imo we could pass the used object to the block, something like the following
```ruby
fields_for_collection do |item|
```
So with this syntaxes you can access to the item and just you need to call I'd to your item object  what do you think? 

Cc @jodosha

**Comment by nerdinand** [comment]
@AlfonsoUceda That sounds like a great solution to me!

**Comment by jodosha** [comment]
@nerdinand Does @AlfonsoUceda suggestion this fixes the problem?

**Comment by nerdinand** [comment]
@jodosha if that got implemented it would fix my problem, yes.

**Comment by jodosha** [comment]
Implemented by #133
